experimental/torch_xla2/test/test_exports.py

```diff
@@ -34,33 +34,112 @@ def setUp(self):
     with torch.no_grad():
       exported = torch.export.export(model, arg)
-      weights, func = torch_xla2.export.exported_program_to_jax(exported)
-      argj = tensor.t2j(arg[0])
-      ans2 = jax.jit(func)(weights, (argj,))[0]
-      ans2 = tensor.j2t(ans2)
-      self.assertTrue(torch.allclose(ans, ans2, atol=1e-5))
+
+    weights, func = torch_xla2.export.exported_program_to_jax(exported)
+    argj = tensor.t2j(arg[0])
+    ans2 = jax.jit(func)(weights, (argj,))[0]
+    ans2 = tensor.j2t(ans2)
+    self.assertTrue(torch.allclose(ans, ans2, atol=1e-5))
+
+    # Convert to StableHLO
+    stablehlo = torch_xla2.export.exported_program_to_stablehlo(exported)
+    module_str = str(stablehlo.mlir_module())
+    self.assertIn("func.func public @main", module_str)
+    self.assertIn("stablehlo.divide", module_str)
+
+  def test_interpolate_dynamic(self):
+    # Export with dynamic dimension constraints on both min and max
+    arg = (torch.randn(3, 3, 200, 200),)
+    model = Interpolate()
+    ans = model(*arg)
+    dynamic_shapes = ({0:  torch.export.Dim("b", min=3, max=10)},)
+
+    with torch.no_grad():
+      exported = torch.export.export(model, arg, dynamic_shapes=dynamic_shapes)
+    stablehlo = torch_xla2.export.exported_program_to_stablehlo(exported)
+    module_str = str(stablehlo.mlir_module())
+
+    # Look for dynamic shape artifacts
+    self.assertIn("func.func public @main(%arg0: tensor<?x3x200x200xf32>", module_str)
+    self.assertIn("stablehlo.dynamic_broadcast_in_dim", module_str)
+    self.assertIn("stablehlo.dynamic_gather", module_str)
+
+  def test_export_dtypes(self):
+    DTYPE_TO_MLIR_STR = {
+      # NO_MAPPING        : jnp.float0 (signless scalar int)
+      torch.bool          : "i1",
+      # NO_MAPPING        : "i4"
+      torch.int8          : "i8",
+      torch.int16         : "i16",
+      torch.int32         : "i32",
+      torch.int64         : "i64",
+      torch.long          : "i64",
+      # NO_MAPPING        : "ui4"
+      torch.uint8         : "ui8",
+      torch.uint16        : "ui16",
+      torch.uint32        : "ui32",
+      torch.uint64        : "ui64",
+      # NO_MAPPING        : "f8E4M3B11FNUZ"
+      torch.float8_e4m3fn : "f8E4M3FN",
+      # NO_MAPPING        : f8E4M3FNUZ
+      torch.float8_e5m2   : "f8E5M2",
+      # NO_MAPPING        : f8E5M2FNUZ
+      torch.bfloat16      : "bf16",
+      torch.half          : "f16",
+      torch.float16       : "f16",
+      torch.float32       : "f32",
+      torch.float64       : "f64",
+      torch.double        : "f64",
+      torch.complex64     : "complex<f32>",
+      torch.complex128    : "complex<f64>",
+      None                : None,
+    }
+
+    model = TensorConstant()
+    for torch_dtype in torch_xla2.tensor.TORCH_DTYPE_TO_JAX.keys():
+      if torch_dtype == None:
+        ## TODO: Figure out what the None mapping should be, seems like:
+        ##   torch.tensor(dtype=None) maps to f32
+        ##   jnp.tensor(dtype=None) maps to f64
```

Comment:
I didn't write the `None:None` mapping, so going to leave this open, in case there's some other implications to mapping.

Also the way the mapping is written now `None->jax.f32` would imply that `jax.f32->None` in the other direction. Technically not wrong but I'd rather directly map to the torch.f32 type as we currently are

Reply:
So in torch, when `dtype` is None it default to `torch.get_default_dtype()` and we have implemented that for tensor constructors here: (https://github.com/pytorch/xla/blob/master/experimental/torch_xla2/torch_xla2/ops/op_base.py#L77)

The above is configurable with `torch.set_default_dtype` and defaults to f32.

Jax has a different configuration: jax_enable_x64 which we enable here: https://github.com/pytorch/xla/blob/master/experimental/torch_xla2/torch_xla2/__init__.py#L7

The Jax configuration's semantics is: by default it's False and everything is at most 32 bits. (meaning, even if doing `x.astype(jnp.float64)`  (same for int64) won't give you float64 but just give you an warning saying it's not enabled. When enabled, you are able to get 64 bit, but also the constructors defaults to 64 bits. 

So the current strategy is to 1. enable jax x64 so at least we can get it if requested. and 2. explicitly read torch.get_default_dtype and pass it on.